R/deploy-site.R

```diff
@@ -116,7 +116,28 @@ deploy_to_branch <- function(pkg = ".",
   dest_dir <- fs::dir_create(fs::file_temp())
   on.exit(fs::dir_delete(dest_dir))
 
-  git("fetch", "origin", branch)
+  tryCatch(
+    git("fetch", "origin", branch),
+
+    system_command_status_error = function(e) {
+      no_remote_branch <- grepl("couldn't find remote ref", e$stderr)
+      if (!no_remote_branch) {
+        stop(e)
+      }
+
+      # If no remote branch, we need to create it
+      git("checkout", "--orphan", branch)
+      git("rm", "-rf", "--quiet", ".")
+      git("commit", "--allow-empty", "-m", sprintf("Initializing %s branch", branch))
+      git("push", "origin", paste0("HEAD:", branch))
+
+      # checkout the previous branch, we assume master.
+      # Unfortunately the @{-1} syntax doesn't seem to work for switching back
+      # from orphan branches :(
+      git("checkout", "master")
```

Comment:
This assumes we were previously on master, which I think should be valid for all our current use cases. I originally wanted to use [git checkout @{-1}](https://stackoverflow.com/a/7207542/2055486) to go back to the previous branch, but it doesn't seem to work properly when you create an orphan branch. :(

Reply:
Would it be safer to record the current branch name so we can switch back? Or could we do it in a worktree that gets abandoned?